compiler/rustc_builtin_macros/src/unreachable.rs

```diff
@@ -0,0 +1,63 @@
+use rustc_ast::ptr::P;
```

Comment:
This file is almost an exact duplicate of `compiler/rustc_builtin_macros/src/panic.rs`. I'd prefer not to duplicate code like this, as it makes maintanance and history tracking more complicated. Instead, we could put both in the same file, and extract the common parts into a common function. E.g. something like:

```rust
pub fn expand_panic<'cx>(cx: &'cx mut ExtCtxt<'_>, sp: Span, tts: TokenStream) -> Box<dyn MacResult + 'cx> {
    let mac = if use_panic_2021(sp) { sym::panic_2021 } else { sym::panic_2015 };
    expand(mac, cx, sp, tts)
}

pub fn expand_unreachable<'cx>(cx: &'cx mut ExtCtxt<'_>, sp: Span, tts: TokenStream) -> Box<dyn MacResult + 'cx> {
    let mac = if use_panic_2021(sp) { sym::unreachable_2021 } else { sym::unreachable_2015 };
    expand(mac, cx, sp, tts)
}

fn expand(mac: Symbol, cx: &'cx mut ExtCtxt<'_>, sp: Span, tts: TokenStream) -> Box<dyn MacResult + 'cx> {
    ...
}
```

Reply:
Done in https://github.com/rust-lang/rust/pull/93179/commits/ac1865fa02e4e2967bed87891ff79be30b9e593e